### DISCUSSION TITLE
[Feature Request] Record custom property to meta tables

### DISCUSSION THREAD

**Discussion by beiifeng** [sequelize/umzug#656] [Ideas] [answered]
Here is a simple  https://github.com/beiifeng/umzug-report

I want to recored `author` or `descript` or some properties to `sequelize_meta` table, but i can implement now.

## Step1  
```base
$ npm i
$ npm run start
```

## Step2

We see the tables was created  
![result-tables](https://github.com/sequelize/umzug/assets/18087934/4eb7f9cd-c72a-4953-a593-601c86b7d893)

and the sequelize_meta data was  
![sequelize-meta](https://github.com/sequelize/umzug/assets/18087934/e88f07ed-4457-4eb9-82c5-1cefc2da542e)

there is no author column.

## What we want
User can override types  
![want-1-override-types](https://github.com/sequelize/umzug/assets/18087934/3f2c4159-0fc8-4b90-8418-fc00c8b63714)

User can use self defined properties with types  
![want-2-add-self-params](https://github.com/sequelize/umzug/assets/18087934/a4b163c4-99ac-48ae-a112-eacbff3c5344)

The properties can taken effected.  
![want-3-can-take-effect](https://github.com/sequelize/umzug/assets/18087934/b92d6a54-247f-4da7-bec5-ad2965ef08a4)

**Comment by mmkal** [comment, marked as answer]
Could you just use context for this?

```ts
class YourStorage implements UmzugStorage<{author: string}> {
  async logMigration({name, context}: MigrationParams<{author: string}>) {
    await this.syncModel()
    await this.model.create({
      name,
      author: context.author,
    })
  }
  async unlogMigration({name, context}: MigrationParams<{author: string}>) {}
  async executed() {
    return []
  }
}
```

**Reply by beiifeng** [reply]
Yes, this a good idea! Thank you so much.
